src/main/java/com/internship/socialnetwork/repository/PostRepository.java

```diff
@@ -0,0 +1,18 @@
+package com.internship.socialnetwork.repository;
+
+import com.internship.socialnetwork.model.Post;
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.Query;
+import org.springframework.data.repository.RepositoryDefinition;
+import org.springframework.data.repository.query.Param;
+import org.springframework.stereotype.Repository;
+
+import java.util.List;
+
+@Repository
+public interface PostRepository extends JpaRepository<Post, Long> {
+
+    @Query("SELECT p FROM Post p WHERE p.author = :author_id")
+    List<Post> findByAuthor(@Param("author_id") Long authorId);
```

Comment:
@Param("author_id") not needed, just use the same value authorId in the query too